File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.js
/**
 * @param {string} s
 * @return {number}
 */
var partitionString = function(s) {
    
    const set = new Set();
    const n = s.length;
    
    let i=0;
    let cnt=0;
    
    while(i<n){
        
        if(set.has(s[i])){
            cnt++;
            set.clear();
            continue;
        }
        
        set.add(s[i]);
        i++;
    }
    
    if(set.size) cnt++;
    return cnt;
};

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        
        int n = s.length();
        
        set<char> st;
        int i=0;
        
        int cnt=0;
        while(i<n){
            
            if(st.find(s[i]) != st.end()){
                cnt++;
                st.clear();
                continue;
            }
            
            st.insert(s[i]);
            i++;
        }
        
        if(st.size() > 0) cnt++;
        
        return cnt;
    }
};